src/app/components/todos-container/todos-container.component.ts

```diff
@@ -1,16 +1,71 @@
-import { Component, OnInit } from '@angular/core';
+import {Component, OnInit} from '@angular/core';
+import {MatDialog} from '@angular/material/dialog'
+import {DialogComponent} from "../dialog/dialog.component";
+import {ITodo} from "../../../../interfaces/ITodo";
+import {EStatuses} from "../../../../enums/Estatus";
+import {TodoService} from "../../services/todo.service";
+import { IUser } from 'interfaces/IUser';
 
 @Component({
   selector: 'app-todos-container',
   templateUrl: './todos-container.component.html',
   styleUrls: ['./todos-container.component.css']
 })
 export class TodosContainerComponent implements OnInit {
-
   title = "Breakthrough TodoApp under Mike's supervision";
-  constructor() { }
+  users?:IUser[]
+
+  constructor(private matDialog:MatDialog,private todoService: TodoService) { 
+     //Get all users in the DB
+     this.todoService.getUsers()
+     .subscribe(
+       data => {
+         this.users = data;
+       },
+       error => {
+         console.log(error)
+       }
+     )
+  }
 
   ngOnInit(): void {
   }
 
+   onOpenDialogClick(){
+    const todo:ITodo = {
+      title: "",
+      assignee: {
+        email: "lukas@gmail.com",
+        name: "Lukas Rasocha"
+      },
+      assigned: {
+        email: "",
+        name: ""
+      },
+      category: 1,
+      status: EStatuses.ASSIGNED,
+      description: "",
+      dateAdded: "",
+      dateCompleted: null
+    }
+
+    const dialogRef = this.matDialog.open(DialogComponent, {
+      data: {emptyTodo:todo, users:this.users},
+      width:"400px",
+      height:"450px"
+    });
+    dialogRef.afterClosed().subscribe(result => {
+    
+     if (typeof result == 'undefined'){
+       return
+     }
+     if (result.cancelled){
+        return
+      } 
+
+      result.emptyTodo.dateAdded = new Date().toISOString().slice(0, 19).replace('T', ' ');
+      this.todoService.create(result.emptyTodo).subscribe()
+    });
+  }
+
 }
```

Comment:
lotso dirtiness. hardcoded values, and I would use a cleaner date formatting solution. 

also vaugeness in the variable naming, line 67 says you create something uising an "empty" todo which im sure is probably false :)